pkg/gadgets/run/types/metadata.go

```diff
@@ -364,136 +139,142 @@ func Populate(m *metadatav1.GadgetMetadata, spec *ebpf.CollectionSpec) error {
 	return nil
 }
 
-func getColumnSize(typ btf.Type) uint {
-	switch typedMember := typ.(type) {
-	case *btf.Int:
-		switch typedMember.Encoding {
-		case btf.Signed:
-			switch typedMember.Size {
-			case 1:
-				return columns.MaxCharsInt8
-			case 2:
-				return columns.MaxCharsInt16
-			case 4:
-				return columns.MaxCharsInt32
-			case 8:
-				return columns.MaxCharsInt64
+func populateTracers(m *metadatav1.GadgetMetadata, spec *ebpf.CollectionSpec) error {
+	tracersInfo, err := getTracerInfo(spec)
+	if err != nil {
+		return err
+	}
 
+	for _, tracerInfo := range tracersInfo {
+		tracerMap := spec.Maps[tracerInfo.mapName]
+		if tracerMap == nil {
+			return fmt.Errorf("map %q not found in eBPF object", tracerInfo.mapName)
+		}
+
+		if err := validateTracerMap(tracerMap); err != nil {
+			return fmt.Errorf("tracer map is invalid: %w", err)
+		}
+
+		var tracerMapStruct *btf.Struct
+		if err := spec.Types.TypeByName(tracerInfo.structName, &tracerMapStruct); err != nil {
+			return fmt.Errorf("finding struct %q in eBPF object: %w", tracerInfo.structName, err)
+		}
+
+		if _, found := m.DataSources[tracerInfo.name]; !found {
+			log.Debugf("Adding datasource %q for tracer with map %q and struct %q",
+				tracerInfo.name, tracerMap.Name, tracerMapStruct.Name)
+
+			m.DataSources[tracerInfo.name] = &metadatav1.DataSource{
+				Fields: make(map[string]metadatav1.Field),
 			}
-		case btf.Unsigned:
-			switch typedMember.Size {
-			case 1:
-				return columns.MaxCharsUint8
-			case 2:
-				return columns.MaxCharsUint16
-			case 4:
-				return columns.MaxCharsUint32
-			case 8:
-				return columns.MaxCharsUint64
-			}
-		case btf.Bool:
-			return columns.MaxCharsBool
-		case btf.Char:
-			return columns.MaxCharsChar
 		}
-	case *btf.Typedef:
-		typ := btfhelpers.GetUnderlyingType(typedMember)
-		return getColumnSize(typ)
+
+		ds := m.DataSources[tracerInfo.name]
+
+		if err := populateDatasourceFields(ds, tracerMapStruct); err != nil {
+			return fmt.Errorf("populating struct: %w", err)
+		}
 	}
 
-	return metadatav1.DefaultColumnWidth
+	return nil
 }
 
-func populateTracers(m *metadatav1.GadgetMetadata, spec *ebpf.CollectionSpec) error {
-	tracerInfo, err := getTracerInfo(spec)
-	if err != nil {
-		return err
-	}
-	if tracerInfo == nil {
-		log.Debug("No tracer found in eBPF object")
-		return nil
+// validateTracerMap only checks if the map type. It does not check the map
+// value name and type because such a information is not available in the map
+// definition for perf event arrays and ring buffers.
```

Comment:
```suggestion
// validateTracerMap only checks the map type. It does not check the map
// name and type because that information is not available in the map
// definition for perf event arrays and ring buffers.
```